### PR TITLE
NEPT-2242: Workbench permissions for Contributor and Editor roles

### PR DESCRIPTION
**Pull request by enriquelacoma**
### Description
Add patch to show content on workbench views when the user has the permission "view all unpublished content"

### Change log

- Added: Patch to change views 

### Commands
 ```drush php-eval 'node_access_rebuild();'```
Or manually on url: "admin/reports/status/rebuild"

### CODE REVIEW

File: resources/patches/workbench_og_grants.patch [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
++ * 
++ *  @return array
++ *    Array with roles ['rid' => 'role name']
++ */ 
++function _workbench_og_view_unpublished_roles() {
++  $defined_roles = user_roles();
++  $authorized_roles = array();
++  foreach($defined_roles as  $key => $value) {
++    $roles = user_role_permissions(array($key => $value));
++    isset($roles[$key]['view all unpublished content']) && $roles[$key]['view all unpublished content'] == 1 ? $authorized_roles[$key] = $value : FALSE;
```
Comment (drupol) [inline resources/patches/workbench_og_grants.patch]: Please use the regular `if () {}` construction.

File: resources/patches/workbench_og_grants.patch [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
++  }
++  return $authorized_roles;
```
Comment (drupol) [inline resources/patches/workbench_og_grants.patch:68]: Missing empty line before return statement.

File: resources/patches/workbench_og_grants.patch [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
++/**
++ * Get roles authorized to see unpublished content.
++ * 
++ *  @return array
++ *    Array with roles ['rid' => 'role name']
++ */ 
++function _workbench_og_view_unpublished_roles() {
++  $defined_roles = user_roles();
++  $authorized_roles = array();
++  foreach($defined_roles as  $key => $value) {
```
Comment (drupol) [inline resources/patches/workbench_og_grants.patch]: Missing empty line before foreach statement.

File: resources/patches/workbench_og_grants.patch [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+       }
+   }
+ }
++/**
++ * Get roles authorized to see unpublished content.
++ * 
++ *  @return array
++ *    Array with roles ['rid' => 'role name']
++ */ 
++function _workbench_og_view_unpublished_roles() {
```
Comment (aritomelo) [inline resources/patches/workbench_og_grants.patch]: The function signature may induce a wrong interpretation, unpublished_roles vs unpublished_content, in this case, the second one is the right, so I would suggest thinking on another signature If I come up with one I let you know!

**Review by aritomelo — CHANGES_REQUESTED** [review]

File: resources/patches/workbench_og_grants.patch (hunk 1 above)
Comment (enriquelacoma) [reply resources/patches/workbench_og_grants.patch]: Changed to if() {}

File: resources/patches/workbench_og_grants.patch (hunk 2 above)
Comment (enriquelacoma) [reply resources/patches/workbench_og_grants.patch:68]: Added new line

File: resources/patches/workbench_og_grants.patch (hunk 4 above)
Comment (enriquelacoma) [reply resources/patches/workbench_og_grants.patch]: Function name changed

File: resources/patches/workbench_og_grants.patch [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
++        // Assign role gid.
++        'gid' => $key,
++        'grant_view' => 1,
++        'grant_update' => 0,
++        'grant_delete' => 0,
++        'priority' => 2,
++      );
++    }
++  }
+   return $grants;
```
Comment (aritomelo) [inline resources/patches/workbench_og_grants.patch:26]: The empty line before the return is missing.

File: resources/patches/workbench_og_grants.patch [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+--- a/workbench_og.module
++++ b/workbench_og.module
+@@ -117,6 +117,22 @@ function workbench_og_node_access_records($node) {
+       );
+     }
+   }
++  else {
++    // Assign grants to unpublished nodes that belong to no group.
++    $roles = _workbench_og_view_authorized_roles();
++    foreach($roles as $key => $value) {
```
Comment (aritomelo) [inline resources/patches/workbench_og_grants.patch]: Missing space between for and '('.

File: resources/patches/workbench_og_grants.patch [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
++    // Assign grant to access unpublished content.
++    $roles = _workbench_og_view_authorized_roles();
++    $enabled_roles = array_intersect($roles, $account->roles);
++    if (count($enabled_roles) > 0) {
++      foreach ($enabled_roles as $key => $value) {
++        $grants['workbench_moderation_og_view_all_unpublished'][] = $key;
++      }
++    }
+   }
+   return $grants;
```
Comment (aritomelo) [inline resources/patches/workbench_og_grants.patch:43]: The empty line before return...

File: resources/patches/workbench_og_grants.patch (hunk 3 above)
Comment (aritomelo) [reply resources/patches/workbench_og_grants.patch]: Missing the space in between foreach and '('

**Review by aritomelo — CHANGES_REQUESTED** [review]
Some code standard to be fixed.

**Review by AlbaParrilla — APPROVED** [review]

**Review by jonhy81 — APPROVED** [review]
Looks good, please proceed.

**Review by aritomelo — APPROVED** [review]
Reviewed and test, OK for me!

**Pull request merged**